engine/common/api_helpers/mixins.py

```diff
@@ -288,25 +289,38 @@ class PreviewTemplateMixin:
     def preview_template(self, request, pk):
         template_body = request.data.get("template_body", None)
         template_name = request.data.get("template_name", None)
+        payload = request.data.get("payload", None)
 
         if template_body is None or template_name is None:
             response = {"preview": None}
             return Response(response, status=status.HTTP_200_OK)
 
         notification_channel, attr_name = self.parse_name_and_notification_channel(template_name)
         if attr_name is None:
-            raise BadRequest(detail={"template_name": "Attr name is required"})
+            raise BadRequest(detail={"template_name": "Template name is missing"})
         if attr_name not in ALL_TEMPLATE_NAMES:
-            raise BadRequest(detail={"template_name": "Unknown attr name"})
+            raise BadRequest(detail={"template_name": "Unknown template name"})
         if attr_name in APPEARANCE_TEMPLATE_NAMES:
             if notification_channel is None:
                 raise BadRequest(detail={"notification_channel": "notification_channel is required"})
             if notification_channel not in NOTIFICATION_CHANNEL_OPTIONS:
                 raise BadRequest(detail={"notification_channel": "Unknown notification_channel"})
 
-        alert_to_template = self.get_alert_to_template()
-        if alert_to_template is None:
-            raise BadRequest(detail="Alert to preview does not exist")
+        if isinstance(self.get_object(), AlertReceiveChannel):
```

Comment:
Original idea was to use get_alert_to_template method from AlertReceiveChannelView and AlertGroupView to not to check instance types here. I propose to pass payload argument to this method and keep logic there.